File: fleet-pay-source-code 1 (1)/fleet-pay-export/app/src/models/company.ts
import { CompanyBusinessHours } from './company_business_hours';
import { NotificationSettings } from './notification_settings';
import { Contact } from './contact';

export interface Company {
  id: string,
  name: string,
  emails?: string[] | null,
  phoneNumbers?: string[] | null,
  profilePicture?: string | null,
  businessHours?: CompanyBusinessHours | null,
  notificationSettings?: NotificationSettings | null,
  accountingDepartmentContacts: Contact[],
  // --- start extra_fields code ---

  // --- end extra_fields code ---
}

File: fleet-pay-source-code 1 (1)/fleet-pay-export/flutter/lib/models/company.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'company_business_hours.dart';
import 'notification_settings.dart';
import 'contact.dart';
// --- start import code ---
  
// --- end import code ---

part 'company.g.dart';
part 'company.freezed.dart';

@unfreezed
class Company with _$Company {
  // --- start init code ---
  
  // --- end init code ---

  factory Company({
    required String id,
    required String name,
    List<String>? emails,
    List<String>? phoneNumbers,
    String? profilePicture,
    CompanyBusinessHours? businessHours,
    NotificationSettings? notificationSettings,
    required List<Contact> accountingDepartmentContacts,
  }) = _Company;

  factory Company.fromJson(Map<String, dynamic> json) => _$CompanyFromJson(json);

  // --- start extension code ---
  
  // --- end extension code ---

}
